File: backend/src/main/java/com/leadgrowth/controller/WorkspaceController.java
package com.leadgrowth.controller;

import com.leadgrowth.dto.AuthResponse;
import com.leadgrowth.dto.CreateWorkspaceRequest;
import com.leadgrowth.dto.JoinWorkspaceRequest;
import com.leadgrowth.dto.WorkspaceUpdateRequest;
import com.leadgrowth.entity.Workspace;
import com.leadgrowth.service.WorkspaceService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/workspaces")
public class WorkspaceController {

    private final WorkspaceService workspaceService;

    public WorkspaceController(WorkspaceService workspaceService) {
        this.workspaceService = workspaceService;
    }

    @PostMapping("/create")
    public ResponseEntity<AuthResponse> createWorkspace(@Valid @RequestBody CreateWorkspaceRequest request) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(workspaceService.createWorkspace(request, email));
    }

    @PostMapping("/join")
    public ResponseEntity<AuthResponse> joinWorkspace(@Valid @RequestBody JoinWorkspaceRequest request) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(workspaceService.joinWorkspace(request, email));
    }

    @GetMapping("/current")
    public ResponseEntity<Workspace> getCurrentWorkspace() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(workspaceService.getCurrentWorkspace(email));
    }

    @PutMapping("/current")
    public ResponseEntity<Workspace> updateWorkspace(@Valid @RequestBody WorkspaceUpdateRequest request) {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return ResponseEntity.ok(workspaceService.updateWorkspace(request, email));
    }

    @DeleteMapping("/current")
    public ResponseEntity<Void> deleteWorkspace() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        workspaceService.deleteWorkspace(email);
        return ResponseEntity.ok().build();
    }
}

File: backend-dotnet/Controllers/WorkspaceController.cs
using System.Security.Claims;
using LeadGrowth.DTOs;
using LeadGrowth.Models;
using LeadGrowth.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace LeadGrowth.Controllers;

[ApiController]
[Route("api/workspaces")]
[Authorize]
public class WorkspaceController : ControllerBase
{
    private readonly IWorkspaceService _workspaceService;

    public WorkspaceController(IWorkspaceService workspaceService)
    {
        _workspaceService = workspaceService;
    }

    [HttpPost]
    [HttpPost("create")]
    public async Task<ActionResult<AuthResponse>> CreateWorkspace([FromBody] CreateWorkspaceRequest request)
    {
        var email = GetUserEmail();
        try
        {
            var response = await _workspaceService.CreateWorkspaceAsync(request, email);
            return Ok(response);
        }
        catch (InvalidOperationException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpPost("join")]
    public async Task<ActionResult<AuthResponse>> JoinWorkspace([FromBody] JoinWorkspaceRequest request)
    {
        var email = GetUserEmail();
        try
        {
            var response = await _workspaceService.JoinWorkspaceAsync(request, email);
            return Ok(response);
        }
        catch (ArgumentException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
        catch (InvalidOperationException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpGet("current")]
    public async Task<ActionResult<Workspace>> GetCurrentWorkspace()
    {
        var email = GetUserEmail();
        try
        {
            var workspace = await _workspaceService.GetCurrentWorkspaceAsync(email);
            return Ok(workspace);
        }
        catch (InvalidOperationException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpPut("current")]
    public async Task<ActionResult<Workspace>> UpdateWorkspace([FromBody] WorkspaceUpdateRequest request)
    {
        var email = GetUserEmail();
        try
        {
            var workspace = await _workspaceService.UpdateWorkspaceAsync(request, email);
            return Ok(workspace);
        }
        catch (ArgumentException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
        catch (InvalidOperationException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    [HttpDelete("current")]
    public async Task<IActionResult> DeleteWorkspace()
    {
        var email = GetUserEmail();
        try
        {
            await _workspaceService.DeleteWorkspaceAsync(email);
            return Ok(new { message = "Workspace deleted successfully" });
        }
        catch (InvalidOperationException ex)
        {
            return BadRequest(new { message = ex.Message });
        }
    }

    private string GetUserEmail()
    {
        return User.FindFirstValue(ClaimTypes.Name) ?? User.FindFirstValue(ClaimTypes.Email) ?? string.Empty;
    }
}
